experimental/emulated_vault/opt/emulated-vault/storage/adapter_base.py

```diff
@@ -0,0 +1,163 @@
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	 http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+
+import sys
+if sys.version_info >= (3, 0):
+	from abc import ABC, abstractmethod	
+else:	
+	ABC = object
+	abstractmethod = lambda f: f
+	
+
+
+class AdapterBase(ABC):
+	"""
+	Base adapter class.
+	This class implements the API required for storing and retriving the content kept in the vault.
+	
+	Methods to be implemented at derived classes:
+	:meth:`get_parameter_storage_path` given a url-key of a parameter return the storage path
+	:meth:`get_parameter_storage_path_from_url_path` given a storage path of a parameter
+	retrun the url-key
+	:meth:`init_cfg` given a config-parser object, read the parameters required for 
+	adapter's operation. Return "success" bool value
+	:meth:`init` prepare the adapter - connecting to storgae / setting the conntent for 
+	"ping" requests etc.. Return "success" bool value
+	:meth:`ping` test the 'ping' status with the adapter. 
+	Return a tuple: "success" bool & "value" kept as ping variable
+	:meth:`read_parameter_by_storage_path` given a storage path retrieve the parameter value.
+	Return a tuple: "success" bool & "value" kept in the parameter
+	:meth:`read_parameters_by_storage_path` given a storage and and a key holding 
+	filters on the key and values.
+	Return "success" bool indicating a sucessful write, and a key->value dictionary 
+	for the relevant parameters
+	:meth:`write_parameter_by_storage_path` given a storage path and a value string,
+	keep the parameter value. Return "success" bool indicating a sucessful write
+	:meth:`remove_parameter_by_storage_path` given a storage path, delete the parameter
+	from the DB. Return "success" bool indicating a sucessful deletion
+	"""
+
+	@abstractmethod
+	def init_cfg(self, fullConfig):# -> bool:
+		"""
+		Method to be implemented at derived classes
+		Initialize the class basic parameters. Part of Adapter required API.
+		:param fullConfig: configuration to operate upon.
+		:type fullConfig: configparser.ConfigParser class
+		:return: 'True' for successful initialization
+		:rtype: bool
+		"""
+		raise NotImplemented()#...
```

Comment:
`NotImplemented` is not callable, it's a language constant. You don't `raise NotImplemented`, you `return NotImplemented`, but that's not appropriate here. I think what you want to do is `raise NotImplementedError()`.